src/figuresClasses.ts

```diff
@@ -1,19 +1,113 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+}
+
+interface ITriangle extends Figure {
+  side1: number,
+  side2: number,
+  side3: number,
+}
+
+interface ICircle extends Figure {
+  radius: number
+}
+
+interface IRectangle extends Figure {
+  width: number,
+  height: number,
+}
 
+function throwErrorOnNegativeValue(variable: number): Error | void {
+  if (variable < 1) {
+    throw new Error("Don't write plz negative numbers");
+  }
 }
 
-export class Triangle {
+export class Triangle implements ITriangle {
+  readonly shape: 'triangle' | 'circle' | 'rectangle';
 
+  constructor(
+    readonly color: 'red' | 'green' | 'blue',
+    readonly side1: number,
+    readonly side2: number,
+    readonly side3: number,
+  ) {
+    throwErrorOnNegativeValue(side1);
+    throwErrorOnNegativeValue(side2);
+    throwErrorOnNegativeValue(side3);
+
+    if (!(side1 + side2 > side3)) {
+      throw new Error();
+    }
+
+    if (!(side2 + side3 > side1)) {
+      throw new Error();
```

Comment:
Error message should be more descriptive. For example: 'The sum of the lengths of two sides of a triangle should be greater than the length of the third side.'